src/ucp/api/ucp.h

```diff
@@ -463,13 +463,21 @@ enum {
  * @ingroup UCP_WORKER
  * @brief Flags for a UCP Active Message callback.
  *
- * Flags that indicate how to handle UCP Active Messages
- * Currently only UCP_AM_FLAG_WHOLE_MSG is supported,
- * which indicates the entire message is handled in one
- * callback.
+ * Flags that indicate how to handle UCP Active Messages.
  */
 enum ucp_am_cb_flags {
-    UCP_AM_FLAG_WHOLE_MSG = UCS_BIT(0)
+    /**
+     * Indicates that the entire message will be handled in one callback.
+     */
+    UCP_AM_FLAG_WHOLE_MSG       = UCS_BIT(0),
+
+    /**
+     * Guarantees that @ref ucp_am_recv_callback_t callback, invoked for the
+     * certain active message id, will always provide persistent pointer to the
+     * data (i.e. UCP_AM_RECV_ATTR_FLAG_DATA will be set in
+     * @a param->recv_attr flags)
+     */
+    UCP_AM_FLAG_PERSISTENT_DATA = UCS_BIT(1)
```

Comment:
description is not clear: what does it mean exactly "persistent data" ? who is responsible to release the pointer and how?

Reply:
it is described for `UCP_AM_RECV_ATTR_FLAG_DATA ` which is mentioned here. Describe it here as well?

Reply:
@yosefe?

Reply:
@yosefe, ping

Reply:
> `/* Guarantees that the specified @ref ucp_am_recv_callback_t callback will always be called with @ref UCP_AM_RECV_ATTR_FLAG_DATA flag set, so the @ref ucp_am_recv_data() could be used to receive the data */`

This PR does not extend semantic of `ucp_am_recv_data()` (which can be called for rndv packets only so far)

Reply:
`/* Guarantees that the specified @ref ucp_am_recv_callback_t callback will always be called with @ref UCP_AM_RECV_ATTR_FLAG_DATA flag set, so the data would be accessible outside the callback, until release data is called */`

Reply:
`until release data is called` please add reference to release call.

Reply:
doc updated